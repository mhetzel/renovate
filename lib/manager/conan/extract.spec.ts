import { loadFixture } from '../../../test/util';
import extractPackageFile from './extract';

jest.mock('../../util/fs');

const conanfile1 = loadFixture('conanfile.txt');
const conanfile2 = loadFixture('conanfile2.txt');
const conanfile3 = loadFixture('conanfile.py');
describe('manager/conan/extract', () => {
  describe('extractPackageFile', () => {
    it('returns null for empty', () => {
      expect(extractPackageFile('nothing here')).toBeNull();
    });
    it('extracts multiple image lines from conanfile.txt', () => {
      const res = extractPackageFile(conanfile1);
      expect(res.deps).toHaveLength(2);
    });
    it('extracts multiple 0 lines from conanfile.txt', () => {
      const res = extractPackageFile(conanfile2);
      expect(res).toBeNull();
    });
    it('extracts multiple image lines from conanfile.py', () => {
      const res = extractPackageFile(conanfile3);
      expect(res.deps).toHaveLength(15);
    });
  });
});
