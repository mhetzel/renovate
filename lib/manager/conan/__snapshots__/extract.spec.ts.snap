// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/conan/extract extractPackageFile extracts multiple image lines from conanfile.py 1`] = `
Array [
  Object {
    "currentValue": "0.1",
    "depName": "pyreq",
    "depType": "python_requires",
    "lookupName": "pyreq/0.1@user/channel",
    "replaceString": "pyreq/0.1@user/channel",
  },
  Object {
    "currentValue": "0.2",
    "depName": "tool_a",
    "depType": "build_requires",
    "lookupName": "tool_a/0.2@user/testing",
    "replaceString": "tool_a/0.2@user/testing",
  },
  Object {
    "currentValue": "0.2",
    "depName": "tool_b",
    "depType": "build_requires",
    "lookupName": "tool_b/0.2@user/testing",
    "replaceString": "tool_b/0.2@user/testing",
  },
  Object {
    "currentValue": "1.0",
    "depName": "req_a",
    "depType": "requires",
    "lookupName": "req_a/1.0@_/_",
    "replaceString": "req_a/1.0",
  },
  Object {
    "currentValue": "2.1",
    "depName": "req_l",
    "depType": "requires",
    "lookupName": "req_l/2.1@otheruser/testing",
    "replaceString": "req_l/2.1@otheruser/testing",
  },
  Object {
    "currentValue": "0.1",
    "depName": "req_b",
    "depType": "requires",
    "lookupName": "req_b/0.1@user/testing",
    "replaceString": "req_b/0.1@user/testing",
  },
  Object {
    "currentValue": "0.2",
    "depName": "req_d",
    "depType": "requires",
    "lookupName": "req_d/0.2@dummy/stable",
    "replaceString": "req_d/0.2@dummy/stable",
  },
  Object {
    "currentValue": "2.1",
    "depName": "req_e",
    "depType": "requires",
    "lookupName": "req_e/2.1@coder/beta",
    "replaceString": "req_e/2.1@coder/beta",
  },
  Object {
    "currentValue": "1.0",
    "depName": "req_c",
    "depType": "requires",
    "lookupName": "req_c/1.0@user/stable",
    "replaceString": "req_c/1.0@user/stable",
  },
  Object {
    "currentValue": "1.0",
    "depName": "req_f",
    "depType": "requires",
    "lookupName": "req_f/1.0@user/stable",
    "replaceString": "req_f/1.0@user/stable",
  },
  Object {
    "currentValue": "3.0",
    "depName": "req_h",
    "depType": "requires",
    "lookupName": "req_h/3.0@other/beta",
    "replaceString": "req_h/3.0@other/beta",
  },
  Object {
    "currentValue": "[>1.0 <1.8]",
    "depName": "req_g",
    "depType": "requires",
    "lookupName": "req_g/[>1.0 <1.8]@user/stable",
    "replaceString": "req_g/[>1.0 <1.8]@user/stable",
  },
  Object {
    "currentValue": "1.2",
    "depName": "req_i",
    "depType": "requires",
    "lookupName": "req_i/1.2@drl/testing",
    "replaceString": "req_i/1.2@drl/testing",
  },
  Object {
    "currentValue": "2.2",
    "depName": "req_i",
    "depType": "requires",
    "lookupName": "req_i/2.2@drl/stable",
    "replaceString": "req_i/2.2@drl/stable",
  },
  Object {
    "currentValue": "1.2",
    "depName": "req_k",
    "depType": "requires",
    "lookupName": "req_k/1.2@drl/testing",
    "replaceString": "req_k/1.2@drl/testing",
  },
  Object {
    "currentValue": "0.1",
    "depName": "tool_win",
    "depType": "build_requires",
    "lookupName": "tool_win/0.1@user/stable",
    "replaceString": "tool_win/0.1@user/stable",
  },
]
`;

exports[`manager/conan/extract extractPackageFile extracts multiple image lines from conanfile.txt 1`] = `
Array [
  Object {
    "currentValue": "1.9.4",
    "depName": "poco",
    "depType": "requires",
    "lookupName": "poco/1.9.4@_/_",
    "replaceString": "poco/1.9.4",
  },
  Object {
    "currentValue": "[~1.2.3, loose=False]",
    "depName": "zlib",
    "depType": "requires",
    "lookupName": "zlib/[~1.2.3, loose=False]@_/_",
    "replaceString": "zlib/[~1.2.3, loose=False]",
  },
  Object {
    "currentValue": "8.62.134",
    "depName": "fake",
    "depType": "requires",
    "lookupName": "fake/8.62.134@test/dev",
    "replaceString": "fake/8.62.134@test/dev",
  },
  Object {
    "currentValue": "[>1.1 <2.1, include_prerelease=True]",
    "depName": "7zip",
    "depType": "build_requires",
    "lookupName": "7zip/[>1.1 <2.1, include_prerelease=True]@_/_",
    "replaceString": "7zip/[>1.1 <2.1, include_prerelease=True]",
  },
  Object {
    "currentValue": "[~1.2.3, loose=False, include_prerelease=True]",
    "depName": "curl",
    "depType": "build_requires",
    "lookupName": "curl/[~1.2.3, loose=False, include_prerelease=True]@test/dev",
    "replaceString": "curl/[~1.2.3, loose=False, include_prerelease=True]@test/dev",
  },
  Object {
    "currentValue": "[>1.1 <2.1]",
    "depName": "boost",
    "depType": "build_requires",
    "lookupName": "boost/[>1.1 <2.1]@_/_",
    "replaceString": "boost/[>1.1 <2.1]",
  },
  Object {
    "currentValue": "[2.8]",
    "depName": "catch2",
    "depType": "build_requires",
    "lookupName": "catch2/[2.8]@_/_",
    "replaceString": "catch2/[2.8]",
  },
  Object {
    "currentValue": "[~=3.0]",
    "depName": "openssl",
    "depType": "build_requires",
    "lookupName": "openssl/[~=3.0]@test/prod",
    "replaceString": "openssl/[~=3.0]@test/prod",
  },
  Object {
    "currentValue": "[>1.1 || 0.8]",
    "depName": "cmake",
    "depType": "build_requires",
    "lookupName": "cmake/[>1.1 || 0.8]@_/_",
    "replaceString": "cmake/[>1.1 || 0.8]",
  },
  Object {
    "currentValue": "[1.2.7 || >=1.2.9 <2.0.0]",
    "depName": "cryptopp",
    "depType": "build_requires",
    "lookupName": "cryptopp/[1.2.7 || >=1.2.9 <2.0.0]@test/local",
    "replaceString": "cryptopp/[1.2.7 || >=1.2.9 <2.0.0]@test/local",
  },
]
`;
